cub/cub/block/block_load.cuh

```diff
@@ -889,7 +889,18 @@ class BlockLoad
     template <typename RandomAccessIterator>
     _CCCL_DEVICE _CCCL_FORCEINLINE void Load(RandomAccessIterator block_src_it, T (&dst_items)[ITEMS_PER_THREAD])
     {
-      LoadDirectBlocked(linear_tid, block_src_it, dst_items);
+      constexpr bool is_pointer_to_type =
+        ::cuda::std::is_pointer_v<RandomAccessIterator>
+        && ::cuda::std::is_same_v<std::remove_cv_t<::cuda::std::remove_pointer_t<RandomAccessIterator>>, T>;
+
+      if (is_pointer_to_type)
+      {
+        InternalLoadDirectBlockedVectorized<LOAD_DEFAULT>(linear_tid, block_src_it, dst_items);
+      }
```

Comment:
```suggestion
      if (::cuda::std::contiguous_iterator<RandomAccessIterator>)
      {
        InternalLoadDirectBlockedVectorized<LOAD_DEFAULT>(linear_tid, ::cuda::std::to_address(block_src_it), dst_items);
      }
```

Reply:
We should probably use `thrust::is_contiguous_iterator` because that also considers all proclaimed iterators.